src/clr-angular/data/datagrid/datagrid-detail.ts

```diff
@@ -16,7 +16,7 @@ import { ClrDatagridDetailHeader } from './datagrid-detail-header';
   // We put the *ngIf on the clrFocusTrap so it doesn't always exist on the page
   // have to test for presence of header for aria-describedby because it was causing unit tests to crash
   template: `
-    <div clrFocusTrap class="datagrid-detail-pane-content" *ngIf="detailService.isOpen" role="dialog"
+    <div [clrFocusTrap]="{local: true}" class="datagrid-detail-pane-content" *ngIf="detailService.isOpen" role="dialog"
```

Comment:
I used `local` because it was localizing the trap instead of putting it at the top/end of the body. Happy to change, but the purpose is not just mouse events, but any focus change (could also be through the app moving focus) to not be greedy and steal it back. Its not just mouse events potentially.